cardano-cli/src/Cardano/CLI/Shelley/Run/Pretty.hs

```diff
@@ -1,107 +1,166 @@
+friendlyTxBodyAllegra
+  (ShelleyMA.TxBody
+    inputs
+    outputs
+    certificates
+    (Shelley.Wdrl withdrawals)
+    txfee
+    validity
+    update
+    adHash
+    _mint -- mint is not used in Allegra, only in Mary+
+    ) =
+  HashMap.fromList
+    [ "inputs" .= inputs
+    , "outputs" .= fmap friendlyTxOutAllegra outputs
+    , "certificates" .= fmap textShow certificates
+    , "withdrawals" .= withdrawals
+    , "fee" .= txfee
+    , "validity interval" .= friendlyValidityInterval validity
+    , "update" .= fmap textShow update
+    , "auxiliary data hash" .= fmap textShow adHash
+    ]
+
+friendlyTxBodyMary
+  :: ShelleyMA.TxBody (ShelleyMAEra 'Mary StandardCrypto) -> Object
+friendlyTxBodyMary
+  (ShelleyMA.TxBody
+    inputs
+    outputs
+    certificates
+    (Shelley.Wdrl withdrawals)
+    txfee
+    validity
+    update
+    adHash
+    mint) =
+  HashMap.fromList
+    [ "inputs" .= inputs
+    , "outputs" .= fmap friendlyTxOutMary outputs
+    , "certificates" .= fmap textShow certificates
+    , "withdrawals" .= withdrawals
+    , "fee" .= txfee
+    , "validity interval" .= friendlyValidityInterval validity
+    , "update" .= fmap textShow update
+    , "auxiliary data hash" .= fmap textShow adHash
+    , "mint" .= mint
+    ]
+
+friendlyValidityInterval :: ShelleyMA.ValidityInterval -> Value
+friendlyValidityInterval
   ShelleyMA.ValidityInterval{invalidBefore, invalidHereafter} =
     object
-      [ "invalid_before" .= invalidBefore
-      , "invalid_hereafter" .= invalidHereafter
+      [ "invalid before" .= invalidBefore
+      , "invalid hereafter" .= invalidHereafter
       ]
+
+friendlyTxOutShelley :: TxOut (Ledger.ShelleyEra StandardCrypto) -> Value
+friendlyTxOutShelley (TxOut addr amount) =
+  Object $ HashMap.insert "amount" (toJSON amount) $ friendlyAddress addr
+
+friendlyTxOutAllegra :: TxOut (ShelleyMAEra 'Allegra StandardCrypto) -> Value
+friendlyTxOutAllegra (TxOut addr amount) =
+  Object $ HashMap.insert "amount" (toJSON amount) $ friendlyAddress addr
+
+friendlyTxOutMary :: TxOut (ShelleyMAEra 'Mary StandardCrypto) -> Value
+friendlyTxOutMary (TxOut addr amount) =
+  Object $ HashMap.insert "amount" (toJSON amount) $ friendlyAddress addr
+
+friendlyAddress :: Addr StandardCrypto -> Object
+friendlyAddress addr =
+  HashMap.fromList $
+    case addr of
+      Addr net cred ref ->
+        [ "address" .=
+            object
+              [ "network" .= net
+              , "credential" .= cred
+              , "stake reference" .= textShow ref
+              , "Bech32" .= addressBech32
+              ]
+        ]
+      AddrBootstrap _ ->
+        ["bootstrap address" .= object ["Bech32" .= String addressBech32]]
+  where
+    addressBech32 =
+      case fromShelleyAddr @Api.ShelleyEra addr of
+        AddressInEra (ShelleyAddressInEra _) a -> serialiseToBech32 a
+        AddressInEra ByronAddressInAnyEra _ -> panic "expected Shelley address"
```

Comment:
You should use `serialiseAddress` in both cases (see `SerialiseAddress`) . Byron addresses still exist in the Shelley era. We still need to be able to render them.